content/blog/2018-06-07-you-probably-dont-need-derived-state.md

```diff
@@ -200,45 +200,46 @@ class EmailInput extends Component {
+Le composant de formulaire parent pourrait alors [utiliser une `ref` pour appeler cette méthode](/docs/glossary.html#refs). ([Cliquez ici pour voir la démo de cette solution.](https://codesandbox.io/s/l70krvpykl))
 
-Refs can be useful in certain cases like this one, but generally we recommend you use them sparingly. Even in the demo, this imperative method is nonideal because two renders will occur instead of one.
+Les références peuvent être utiles dans certains cas comme celui-ci mais, généralement, nous recommandons de les utiliser avec parcimonie. Même dans le cadre de cette démo, cette méthode impérative est sous-optimale car elle entraine la réalisation de deux rendus au lieu d'un seul.
 
 -----
 
-### Recap {#recap}
+### Récapitulons {#recap}
 
-To recap, when designing a component, it is important to decide whether its data will be controlled or uncontrolled.
+Quand on définie un composant, il est important de décider si ses données seront controlées ou non-controlées.
 
-Instead of trying to **"mirror" a prop value in state**, make the component **controlled**, and consolidate the two diverging values in the state of some parent component. For example, rather than a child accepting a "committed" `props.value` and tracking a "draft" `state.value`, have the parent manage both `state.draftValue` and `state.committedValue` and control the child's value directly. This makes the data flow more explicit and predictable.
+Plutôt que d'essayer de **refléter les valeur des props dans l'état**, faite un composant controlé et réconciliez les valeurs divergentes dans l'état d'un composant parent. Par example, plutôt que d'avoir un composant enfant qui accepterai une valeur « imperative » `props.value` et de suivre les valeurs intermédiaire via `state.value`, faites en sorte que le composant parent gère à la fois `state.draftValue` et `state.committedValue` pour qu'il contrôle ensuite directement la valeur du composant enfant. Ça rend le flux de données plus explicite et prédictif.
 
-For **uncontrolled** components, if you're trying to reset state when a particular prop (usually an ID) changes, you have a few options:
-* **Recommendation: To reset _all internal state_, use the `key` attribute.**
-* Alternative 1: To reset _only certain state fields_, watch for changes in a special property (e.g. `props.userID`).
-* Alternative 2: You can also consider fall back to an imperative instance method using refs.
+Pour les composants non-controlés, si vous essayez de réinitialiser l'état quand une prop particulière (en général un ID) change, vous avez peu d'options :
+* **Recommandation: Réinitialiser _l'intégralité de l'état interne_ en utilisant un attribut `key`.**
+* Alternative 1: Réinitialiser _certain champs de l'état_ en observant les changement d'une propriété donnée (i.e. `props.userID`).
```

Comment:
* Alternative 1: To reset _only certain state fields_, watch for changes in a special property (e.g. `props.userID`).

```suggestion
* Alternative 1 : réinitialiser _certains champs de l'état_ en observant les changements d'une prop donnée (i.e. `props.userID`).
```